src/tracked_struct.rs

```diff
@@ -220,12 +220,18 @@ impl Clone for IdentityMap {
             table: self.table.clone(),
         }
     }
-    fn clone_from(&mut self, source: &Self) {
-        self.table.clone_from(&source.table);
-    }
 }
 
 impl IdentityMap {
+    pub(crate) fn clone_from_slice(&mut self, source: &[(Identity, Id)]) {
+        self.table.clear();
+        self.table.reserve(source.len(), |(k, _)| k.hash);
```

Comment:
I'm not sure if there's a more elegant way of doing this

Reply:
`self.table.extend(source)` should work.